src/herder/test/HerderTests.cpp

```diff
@@ -5706,3 +5707,416 @@ TEST_CASE("SCP message capture from previous ledger", "[herder]")
+        [&]() {
+            // Collect info about orgs
+            ValidatorQuality maxQuality;
+            std::unordered_map<std::string, ValidatorQuality> orgQualities;
+            std::unordered_map<std::string, int> orgSizes;
+            collectOrgInfo(maxQuality, orgQualities, orgSizes, validators);
+
+            // Check per-validator weights
+            HerderImpl& herder = dynamic_cast<HerderImpl&>(app->getHerder());
+            std::unordered_map<std::string, double> normalizedOrgWeights;
+            for (ValidatorEntry const& validator : validators)
+            {
+                uint64_t weight = herder.getHerderSCPDriver().getNodeWeight(
+                    validator.mKey, cfg.QUORUM_SET, false);
+                double normalizedWeight =
+                    static_cast<double>(weight) / UINT64_MAX;
+                normalizedOrgWeights[validator.mHomeDomain] += normalizedWeight;
+
+                std::string const& org = validator.mHomeDomain;
+                REQUIRE_THAT(normalizedWeight,
+                             Catch::Matchers::WithinAbs(
+                                 expectedNormalizedWeight(maxQuality,
+                                                          orgQualities.at(org),
+                                                          orgSizes.at(org)),
+                                 0.0001));
+            }
+
+            // Check per-org weights
+            for (auto const& [org, weight] : normalizedOrgWeights)
+            {
+                REQUIRE_THAT(weight, Catch::Matchers::WithinAbs(
+                                         expectedOrgNormalizedWeight(
+                                             maxQuality, orgQualities.at(org)),
+                                         0.0001));
+            }
+        });
+}
+
+// Test that HerderSCPDriver::getNodeWeight produces weights that result in a
+// fair distribution of nomination wins.
+TEST_CASE_VERSIONS("getNodeWeight", "[herder]")
+{
+    SECTION("3 tier 1 validators, 1 org")
+    {
+        testWeights(simpleThreeNode().second);
+    }
+
+    SECTION("11 tier 1 validators, 3 unbalanced orgs")
+    {
+        testWeights(unbalancedOrgs().second);
+    }
+
+    SECTION("Random topology")
+    {
+        // Test weights for 1000 random topologies of up to 200 validators
+        for (int i = 0; i < 1000; ++i)
+        {
+            testWeights(randomTopology(200).second);
+        }
+    }
+}
+
+// Spin up a simulation of validators and run for `numLedgers`. After running,
+// check that the win percentages of each node and org are within 5% of the
+// expected win percentages. On large runs with higher `numLedgers` counts, this
+// can be quite slow. It's a more thorough test of the fairness of the
+// nomination algorithm than the `getNodeWeight` test, but at the expense of not
+// being capable of testing as many configurations due to the time it takes to
+// run.
+static void
+testWinProbabilities(std::vector<SecretKey> const& sks,
```

Comment:
Do we need full Simulation here? We should be able to just run leader election (similar to SCP tests) on its own without the rest of consensus (which I think is why the test is taking a long time).

Reply:
I rewrote this test to remove `Simulation` and just test `NomiationProtocol`. In doing so I also bumped up the number of slots tested for each scenario to 10,000.

Reply:
nice, thanks! Did it speed up the test?

Reply:
It made a huge difference! A single simulation used to take up to 2 hours, but now the test runs 13 simulations in under a minute.